feature/notification/impl/src/commonMain/kotlin/net/thunderbird/feature/notification/impl/sender/DefaultNotificationSender.kt

```diff
@@ -2,41 +2,72 @@ package net.thunderbird.feature.notification.impl.sender
+ * and emits the result of the execution as a [Flow].
  *
- * @param commandFactory The factory used to create notification commands.
+ * @param logger The logger instance for logging events.
+ * @param featureFlagProvider Provider for accessing feature flag states.
+ * @param notificationRegistry Registry for managing notifications.
+ * @param systemNotificationNotifier Notifier specifically for system notifications.
+ * @param inAppNotificationNotifier Notifier specifically for in-app notifications.
  */
 class DefaultNotificationSender internal constructor(
-    private val commandFactory: NotificationCommandFactory,
+    private val logger: Logger,
+    private val featureFlagProvider: FeatureFlagProvider,
+    private val notificationRegistry: NotificationRegistry,
+    private val systemNotificationNotifier: NotificationNotifier<SystemNotification>,
+    private val inAppNotificationNotifier: NotificationNotifier<InAppNotification>,
 ) : NotificationSender {
-    /**
-     * Sends a notification by creating and executing the appropriate commands.
-     *
-     * This function takes a [Notification] object, uses the [commandFactory] to generate
-     * a list of [NotificationCommand]s tailored to that notification, and then executes
-     * each command sequentially. The result of each command execution ([NotificationCommand.CommandOutcome])
-     * is emitted as part of the returned [Flow].
-     *
-     * @param notification The [Notification] to be sent.
-     * @return A [Flow] that emits the [NotificationCommand.CommandOutcome] for each executed command.
-     */
     override fun send(notification: Notification): Flow<Outcome<Success<Notification>, Failure<Notification>>> = flow {
-        val commands = commandFactory.create(notification)
+        val commands = buildCommands(notification)
         commands.forEach { command ->
             emit(command.execute())
         }
     }
+
+    private fun buildCommands(notification: Notification): List<NotificationCommand<out Notification>> = buildList {
+        if (notification is SystemNotification) {
+            add(
+                DisplaySystemNotificationCommand(
+                    logger = logger,
+                    featureFlagProvider = featureFlagProvider,
+                    notificationRegistry = notificationRegistry,
+                    notification = notification,
+                    notifier = systemNotificationNotifier,
+                ),
+            )
+        }
+
+        if (notification is InAppNotification) {
+            add(
+                DisplayInAppNotificationCommand(
+                    logger = logger,
+                    featureFlagProvider = featureFlagProvider,
+                    notificationRegistry = notificationRegistry,
+                    notification = notification,
+                    notifier = inAppNotificationNotifier,
+                ),
+            )
+        }
+    }
```

Comment:
What should happen when this is called with an unknown notification type? I think this should log the unknown case.